style_guides/js_style_guide.md

```diff
@@ -608,158 +729,97 @@ function loadUser(id) {
   ...
 }
 
-var isSessionValid = (session.expires < Date.now());
+const isSessionValid = (session.expires < Date.now());
 if (isSessionValid) {
   ...
 }
-```
 
-*Wrong:*
+// bad
 
-```js
 // Execute a regex
-var matches = item.match(/ID_([^\n]+)=([^\n]+)/));
+const matches = item.match(/ID_([^\n]+)=([^\n]+)/));
 
 // Usage: loadUser(5, function() { ... })
 function loadUser(id, cb) {
   // ...
 }
 
 // Check if the session is valid
-var isSessionValid = (session.expires < Date.now());
+const isSessionValid = (session.expires < Date.now());
 // If the session is valid
 if (isSessionValid) {
-  // ...
+  ...
 }
 ```
 
 ## Do not comment out code
 
-We use a version management system. If a line of code is no longer needed, remove it, don't simply comment it out.
+We use a version management system. If a line of code is no longer needed,
+remove it, don't simply comment it out.
 
 ## Classes/Constructors and Inheritance
 
-While JavaScript it is not always considered an object-oriented language, it does have the building blocks for writing object oriented code. Of course, as with all things JavaScript, there are many ways this can be accomplished. Generally, we try to err on the side of readability.
-
-### Capitalized function definition as Constructors
-
-When Defining a Class/Constructor, use the function definition syntax.
-
-*Right:*
-```js
-function ClassName() {
-
-}
-```
-
-*Wrong:*
-```js
-var ClassName = function () {};
-```
-
-### Inheritance should be done with a utility
-
-While you can do it with pure JS, a utility will remove a lot of boilerplate, and be more readable and functional.
-
-*Right:*
+In general, use of the `new` operator is discouraged. If you must use a
```

Comment:
Instead of getting rid of the sentence, how about adding a "better" section between "best" and "good" showing an example of a factory function, and relocate this sentence to that section? In addition to explaining the downsides of `new` (makes refactoring much more challenging, exposes object instantiation details of the class to the world) we could also show how to create private member properties.

```
// better
function usersFactory () {
  const users = [];

  return Object.create({
    add (user) {
      users.push(user);
      return [...users];
    }
  });
}
```